source/codegen/common_helpers.py

```diff
@@ -1207,18 +1207,58 @@ def get_params_needing_initialization(parameters: List[dict]) -> List[dict]:
 def filter_moniker_streaming_functions(functions, functions_to_generate):
     """Return streaming functions that need to be generated."""
     return [
-        name for name in functions_to_generate if functions[name].get("is_streaming_api", False)
+        name for name in functions_to_generate if is_moniker_streaming_function(functions[name])
     ]
 
 
+def is_moniker_streaming_function(function):
+    """Whether this function is for streaming data through moniker."""
+    return function.get("is_streaming_api", False)
+
+
 def get_data_moniker_function_name(function_name):
     """Return the corresponding moniker function name for the given C API function."""
     return function_name.replace("Begin", "Moniker")
 
 
-def get_data_moniker_struct_name(function_name):
-    """Return the corresponding moniker function name for the given C API function."""
-    return f"{function_name.replace('Begin', 'Moniker')}Data"
+def get_data_moniker_struct_name(begin_function_name):
```

Comment:
Can you go through the new/updated methods and add type hints for the inputs and output? It helps see at a glance what's expected of the methods.

Reply:
Added typehints.